core/coreapi/interface/swarm.go

```diff
@@ -0,0 +1,52 @@
+package iface
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	ma "gx/ipfs/QmYmsdtJ3HsodkePE3eU3TsCaP2YvPZJ4LoXnNkDE5Tpt7/go-multiaddr"
+	"gx/ipfs/QmZNkThpqfVXs9GNbexPrfBbXSLNYeKrE7jwFM2oqHbyqN/go-libp2p-protocol"
+	"gx/ipfs/QmbNepETomvmXfz1X5pHNFD2QuPqnqi47dTd94QJWSorQ3/go-libp2p-peer"
+	pstore "gx/ipfs/QmfAQMFpgDU2U4BXG64qVr8HSiictfWvkSBz7Y2oDj65st/go-libp2p-peerstore"
+	net "gx/ipfs/QmfDPh144WGBqRxZb1TGDHerbMnZATrHZggAPw7putNnBq/go-libp2p-net"
+)
+
+var (
+	ErrNotConnected = errors.New("not connected")
+	ErrConnNotFound = errors.New("conn not found")
+)
+
+// ConnectionInfo contains information about a peer
+type ConnectionInfo interface {
+	// ID returns PeerID
+	ID() peer.ID
+
+	// Address returns the multiaddress via which we are connected with the peer
+	Address() ma.Multiaddr
+
+	// Direction returns which way the connection was established
+	Direction() net.Direction
```

Comment:
Any reason to have this over `Stat`? The nice thing about `Stat` is that it's extensible.

Reply:
Assuming you mean `go-libp2p-net.Stat` - `ipfs swarm peers` doesn't return info from the `Extra` map, so we would have to expose it somehow for a good implementation in go-ipfs-api.

Reply:
Got it. Yeah, that may be tricky as it could have arbitrary data. We can punt on that.